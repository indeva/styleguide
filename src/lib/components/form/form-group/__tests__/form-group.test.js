import React from 'react';
import renderer from 'react-test-renderer';
import FormGroup from '..';

describe('FormGroup', () => {
  describe('when there is not any prop', () => {
    it('renders the FormGroup', () => {
      const component = renderer.create(
        <FormGroup>
          <div />
        </FormGroup>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when the errorMessage prop has been filled', () => {
    it('renders the FormGroup with error', () => {
      const errorMessage = 'test';
      const component = renderer.create(
        <FormGroup errorMessage={errorMessage}>
          <div />
        </FormGroup>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
