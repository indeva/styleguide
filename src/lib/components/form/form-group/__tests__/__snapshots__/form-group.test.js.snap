// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup when the errorMessage prop has been filled renders the FormGroup with error 1`] = `
<div
  className="form-group"
>
  <div />
  <span
    className="form-group__message form-group__message--error"
  >
    test
  </span>
</div>
`;

exports[`FormGroup when there is not any prop renders the FormGroup 1`] = `
<div
  className="form-group"
>
  <div />
  <span
    className="form-group__message"
  />
</div>
`;
